modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -0,0 +1,212 @@
+                .thenReturn("https://localhost:9443/scim2/Users");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getUser(id, requiredAttributes)).thenReturn(null);
+
+        SCIMResponse outputScimResponse = userResourceManager.get(id, userManager, attributes, excludeAttributes);
+
+        Assert.assertNull(outputScimResponse);
+    }
+
+    @DataProvider(name = "dataForListWithGetInt")
+    public Object[][] dataToGetListInt() throws CharonException, BadRequestException, InternalErrorException {
+
+        getNewUser();
+        UserManager userManager = mock(UserManager.class);
+
+        return new Object[][]{
+
+                {userManager, null, 1, 2, null, null, "PRIMARY", "emails", null, 200}
+        };
+    }
+
+    @Test(dataProvider = "dataForListWithGetInt")
+    public void testListWithGetInt(Object objectUserManager, String filter, int startIndexInt, int countInt,
+                                   String sortBy, String sortOrder, String domainName, String attributes,
+                                   String excludeAttributes, int expectedScimResponseStatus) {
+
+        UserManager userManager = (UserManager) objectUserManager;
+
+        SCIMResponse outputScimResponse = userResourceManager.listWithGET(userManager, filter, startIndexInt,
+                countInt, sortBy, sortOrder, domainName, attributes, excludeAttributes);
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForListWithGetInteger")
+    public Object[][] dataToGetListInteger() throws CharonException, BadRequestException, InternalErrorException {
+
+        getNewUser();
+        UserManager userManager = mock(UserManager.class);
+
+        return new Object[][]{
+
+                {userManager, null, 1, 2, null, null, "PRIMARY", "emails", null, 200},
+                {userManager, "userName sw Rash", 1, 2, null, null, "PRIMARY", "userName,name.familyName",
+                        "emails", 200}
+        };
+    }
+
+    @Test(dataProvider = "dataForListWithGetInteger")
+    public void testListWithGetInteger(Object objectUserManager, String filter, Integer startIndexInt,
+                        Integer countInt, String sortBy, String sortOrder, String domainName,
+                        String attributes, String excludeAttributes, int expectedScimResponseStatus) {
+
+        UserManager userManager = (UserManager) objectUserManager;
+
+        SCIMResponse outputScimResponse = userResourceManager.listWithGET(userManager, filter, startIndexInt,
+                countInt, sortBy, sortOrder, domainName, attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
```

Comment:
unnecessary new line.

Reply:
fixed in 46de98fc2140524b63a2fc404dda55b17540ecce